src/streaming/controllers/BufferController.js

```diff
@@ -559,6 +559,24 @@ function BufferController(config) {
             if (currentTimeRequest) {
                 rangeStart = Math.max(currentTimeRequest.startTime + currentTimeRequest.duration, rangeStart);
             }
+
+            // Never remove the contiguous range of targetTime in order to avoid flushes & reenqueues in WebKit based browsers
+            const ua = navigator.userAgent.toLowerCase();
```

Comment:
Please hide all this functionality behind a settings flag, e.g. `buffer.enableEmbeddedWebkitPruning` or sth similar

Reply:
This is a really good idea. Thanks!